### PR TITLE
Upgrade gaia release on CI and allow multiple e2e tests (current and future release)

### PR DESCRIPTION
**Pull request by andynog**
Closes: #1408 

## Description

Added logic to run e2e tests in CI with current gaia release (v5.0.8) and future gaia releases 
______

For contributor use:

- [ ] Added a changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] If applicable: Unit tests written, added test to CI.
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by andynog** [comment]
the md-link-check seems to be failing but I think the links are valid to me, maybe the check is just getting 502 error because it tries to retrieve the links and cannot ?

**Comment by andynog** [comment]
Actually the Gaia v6 e2e test failed even though it shows green. It failed on CI too with the same error I got in my local machine. Probably just didn’t error out because I set a flag to continue on errors on the workflow file

```
 - name: Run relayer workflow
        continue-on-error: true
        run: docker exec relayer /bin/sh -c /relayer/e2e.sh
```
I think that’s the only way I guess to prevent it to error and block the checks, found it on this long thread https://github.com/actions/toolkit/issues/399

**Comment by romac** [comment]
If I understand correctly, setting `continue-on-error` to `true` should still mark the job as failed no?

File: .github/workflows/e2e-gaia-current-release.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: End to End testing
+name: End to End testing (Current Gaia)
```
Comment (romac) [inline .github/workflows/e2e-gaia-current-release.yaml:1]: Can we put the actual Gaia version here?

File: .github/workflows/e2e-gaia-future-release.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: End to End testing (Future Gaia)
```
Comment (romac) [inline .github/workflows/e2e-gaia-future-release.yaml:1]: Same here.

Comment (andynog) [reply .github/workflows/e2e-gaia-future-release.yaml]: Done

File: .github/workflows/e2e-gaia-current-release.yaml (hunk 1 above)
Comment (andynog) [reply .github/workflows/e2e-gaia-current-release.yaml]: Done

**Comment by romac** [comment]
@andynog can we update this to Gaia v6.0.0-rc3? https://github.com/cosmos/gaia/releases/tag/v6.0.0-rc3

**Comment by andynog** [comment]
> @andynog can we update this to Gaia v6.0.0-rc3? https://github.com/cosmos/gaia/releases/tag/v6.0.0-rc3

OK, will look into that @romac

**Comment by romac** [comment]
Thank you @andynog 🌷

**Comment by romac** [comment]
Yay, looks like the End-to-End testing job is now passing again with Gaia v6.0.0-rc3 :)

**Review by romac — APPROVED** [review]

**Comment by romac** [comment]
Ran into the same spurious test failure we were having with rc2 in #1576, will be fixed by https://github.com/informalsystems/ibc-rs/pull/1576/commits/9c587770b38e2d840120f7dc3efe3aeb65db8059

**Pull request merged**